library.js

```diff
@@ -145,141 +191,116 @@ plugin.verifyUser = function(uid, callback) {
 	});
 };
 
-plugin.findUser = function(payload, callback) {
-	// If payload id resolves to a user, return the uid, otherwise register a new user
-	winston.verbose('[session-sharing] Payload verified');
-
-	var parent = plugin.settings['payload:parent'],
-		id = parent ? payload[parent][plugin.settings['payload:id']] : payload[plugin.settings['payload:id']],
-		email = parent ? payload[parent][plugin.settings['payload:email']] : payload[plugin.settings['payload:email']],
-		username = parent ? payload[parent][plugin.settings['payload:username']] : payload[plugin.settings['payload:username']],
-		firstName = parent ? payload[parent][plugin.settings['payload:firstName']] : payload[plugin.settings['payload:firstName']],
-		lastName = parent ? payload[parent][plugin.settings['payload:lastName']] : payload[plugin.settings['payload:lastName']],
-		picture = parent ? payload[parent][plugin.settings['payload:picture']] : payload[plugin.settings['payload:picture']];
-
-	var fullname = [firstName, lastName].join(' ').trim();
-
-	if (!username && firstName && lastName) {
-		username = fullname;
-	} else if (!username && firstName && !lastName) {
-		username = firstName;
-	} else if (!username && !firstName && lastName) {
-		username = lastName;
+plugin.findOrCreateUser = function(userData, continueAfterFindOrCreate) {
```

Comment:
Particular, like in, I am not used to it naming it anything. Since you can call a callback from within any nested closure, that may all have their own callback-parameters, I just deemed it makes sense. It also indicates that it is a continuation and not a callback per se, which I conventionally perceive as a function being called, to give back information about global state changes that have been made. Never mind, I am not an node developer! :)